apps/dashboard/app/models/posix_file.rb

```diff
@@ -62,10 +62,14 @@ def to_h
       owner:        PosixFile.username_from_cache(stat.uid),
       mode:         stat.mode,
       dev:          stat.dev,
-      downloadable: !pipe? && readable?
+      downloadable: downloadable?
     }
   end
 
+  def downloadable?
+    file? && readable?
```

Comment:
I didn't figure out how the download button works, but should `directory?` be involved in this check, to enable zipping and downloading a directory?

Reply:
Good catch! Apparently the tests use the download button, so I didn't notice.